Common/Securities/Volatility/BaseVolatilityModel.cs

```diff
@@ -56,6 +62,56 @@ public virtual void Update(Security security, BaseData data)
         {
         }
 
+        /// <summary>
+        /// Applies a dividend to the portfolio
+        /// </summary>
+        /// <param name="dividend">The dividend to be applied</param>
+        /// <param name="liveMode">True if live mode, false for backtest</param>
+        /// <param name="mode">The <see cref="DataNormalizationMode"/> for this security</param>
+        public virtual void ApplyDividend(Dividend dividend, bool liveMode, DataNormalizationMode mode)
+        {
+            // only apply splits in live or raw data mode
+            if (!liveMode && !(mode == DataNormalizationMode.Raw || mode == DataNormalizationMode.SplitAdjusted))
+            {
+                return;
+            }
+
+            var factor = 1 - dividend.Distribution / dividend.ReferencePrice;
+            SetLastFactor(factor);
+        }
+
+        /// <summary>
+        /// Applies a split to the model
+        /// </summary>
+        /// <param name="split">The split to be applied</param>
+        /// <param name="liveMode">True if live mode, false for backtest</param>
+        /// <param name="mode">The <see cref="DataNormalizationMode"/> for this security</param>
+        public virtual void ApplySplit(Split split, bool liveMode, DataNormalizationMode mode)
+        {
+            // only apply splits in live or raw data mode
+            if (!liveMode && mode != DataNormalizationMode.Raw)
+            {
+                return;
+            }
+
+            SetLastFactor(split.SplitFactor);
+        }
+
+        /// <summary>
+        /// Sets the value of the latest price factor
+        /// </summary>
+        /// <param name="factor">Latest price factor to be applied</param>
+        public void SetLastFactor(decimal? factor)
```

Comment:
This should likely be placed inside the property's setter to enforce encapsulation of the property's data.

Reply:
Also, consider making `LastFactor` private/protected -- I don't think consumers should require/be coupled to this data